ip-whitelist/cmd/resource/resource.go

```diff
@@ -0,0 +1,247 @@
+package resource
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectID := *currentModel.ProjectId.Value()
+
+	request := getProjectIPWhitelistRequest(currentModel)
+
+	_, _, err = client.ProjectIPWhitelist.Create(context.Background(), projectID, request)
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	var withelist []string
+	whiteListMap(request, func(entry string) {
+		withelist = append(withelist, entry)
+	})
+
+	id := encodeStateID(map[string]string{
```

Comment:
Could you explain to me the technical rationale of why we encode the id and the entries into a giant string? It's a weird approach so just trying to understand why we do it.

Reply:
It is just to keep track of all entries in case one is removed/modified on the state. And we also need an ID, so we are storing it there. This is the same case with terraform.